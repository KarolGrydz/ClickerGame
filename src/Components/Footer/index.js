import React from 'react';
import { Col } from 'reactstrap';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Modals from '../Modals';

const Footer = players => (
  <FooterDiv>
    <Col md="12" sm="12" xs="12" className="text-center">
      <Modals players={players} />
    </Col>
    <a href="https://kg.glogow.pl">English version</a>
    <p>
      Stworzone przez <a href="https://kg.glogow.pl">Ranald</a> 2016, Wszelkie
      prawa zastrze≈ºone
    </p>
  </FooterDiv>
);

Footer.propTypes = {
  players: PropTypes.array.isRequired
};

export default Footer;

const FooterDiv = styled.div`
  padding-top: 50px;
  font-size: 0.75rem;
  background-color: #fff;
`;
