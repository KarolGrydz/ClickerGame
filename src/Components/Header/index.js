import React from 'react';
import { Paragraf, Title } from '../App';

const AppHeader = ({ addScore }) => {
  if (addScore) {
    return <Title>Dodaj swój wynik</Title>;
  } else {
    return (
      <div>
        <Title>Gra w klikacza</Title>
        <Paragraf>
          Ile razy jesteś w stanie kliknąć w przeciągu 10 sekund?
          <br />
          Zabawa zaczyna się od pierwszego kliknięcia...
        </Paragraf>
      </div>
    );
  }
};

export default AppHeader;
