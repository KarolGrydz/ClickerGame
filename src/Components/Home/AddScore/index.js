import React, { Component } from 'react';
import {
  Button,
  Form,
  FormGroup,
  Input,
  Container,
  Col,
  Row
} from 'reactstrap';

class AddScore extends Component {
  state = {
    counts: this.props.numberOfClickToAdd,
    playerName: ''
  };

  handleChange = input => e => {
    this.setState({ [input]: e.target.value });
  };

  handleEnter = e => {
    if (e.key === 'Enter') {
      return this.addToAppjs(e);
    }
  };

  addToAppjs = e => {
    e.preventDefault();
    const { playerName, counts } = this.state;
    const { addScoreToFireBase, changeSite } = this.props;
    addScoreToFireBase(playerName, counts);
    return changeSite();
  };

  render() {
    const { counts } = this.state;
    return (
      <Container>
        <Row>
          <Form>
            <FormGroup>
              <Col md={{ size: 8, offset: 2 }}>
                <Input
                  type="text"
                  name="name"
                  placeholder="Wpisz swoje imię"
                  onChange={this.handleChange('playerName')}
                  onKeyPress={this.handleEnter}
                />
              </Col>
            </FormGroup>
            <Col md="12">
              <h3>Twoje punkty {counts}</h3>
            </Col>
            <Col md="12">
              <Button onClick={this.addToAppjs}>Zatwierdź</Button>
            </Col>
          </Form>
        </Row>
      </Container>
    );
  }
}

export default AddScore;
