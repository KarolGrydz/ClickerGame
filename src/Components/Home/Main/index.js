import React, { Component } from 'react';
import { Button } from 'reactstrap';
import { Paragraf } from '../../App';

import { endGame } from '../../../helperFunctions/endGame';
import Counter from '../../Counter';

class Main extends Component {
  state = {
    count: 0,
    mili: 0,
    sec: 0,
    startStopper: true
  };

  toggleClick = () => {
    this.setState({
      count: this.state.count + 1
    });

    if (this.state.startStopper) {
      this.toggleTime();
      return this.setState({
        startStopper: false
      });
    }
  };

  toggleTime = () => {
    let sec = 0;
    let mili = 0;
    let total = 0;

    this.timer = setInterval(() => {
      total += 1;
      sec = Math.floor(total / 100);
      mili = total % 100;
      this.setState({ mili, sec });
      if (total > 999) {
        this.setState({
          startStopper: true
        });
        clearInterval(this.timer);
        return this.stopTime();
      }
    }, 10);
  };

  componentWillUnmount() {
    clearInterval(this.timer);
  }

  checkToAddScoreToHighScores = () => {
    let { players } = this.props;
    const { addNewScore, changeSite } = this.props;
    const { count } = this.state;
    this.setState({ count: 0, mili: 0, sec: 0 });
    for (let player of players) {
      if (count >= player.points) {
        return [addNewScore(count), changeSite()];
      }
    }
  };

  stopTime = () => {
    const { count } = this.state;

    endGame(count);
    return this.checkToAddScoreToHighScores();
  };

  render() {
    const { count, mili, sec } = this.state;
    return (
      <React.Fragment>
        <Button color="primary" size="lg" onClick={this.toggleClick}>
          Kliknij mnie!
        </Button>
        <Paragraf>
          Kliknełeś mnie! <Counter value={count} /> razy!
        </Paragraf>
        <Paragraf>
          <Counter value={sec} />:<Counter value={mili} />
        </Paragraf>
      </React.Fragment>
    );
  }
}

export default Main;
