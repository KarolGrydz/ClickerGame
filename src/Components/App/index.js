import React, { Component } from 'react';
import { Container, Col } from 'reactstrap';
import styled from 'styled-components';
import Footer from '../Footer';
import Header from '../Header';
import Home from '../Home';
import Firebase from '../Firebase';

import 'bootstrap/dist/css/bootstrap.min.css';
import './App.css';

class App extends Component {
  state = {
    players: [],
    addScore: false,
    numberOfClickToAdd: 0,
    playerExist: true,
    firebase: new Firebase()
  };

  addNewScore = numberOfClickToAdd => this.setState({ numberOfClickToAdd });

  changeSite = () => {
    let { addScore } = this.state;
    addScore = !addScore;
    this.setState({ addScore });
  };

  gotData = data => {
    let newPlayers = [];
    let scores = data.val();
    let keys = Object.keys(scores);
    for (let i = 0; i < keys.length; i++) {
      let k = keys[i];
      let name = scores[k].name;
      let points = scores[k].points;
      let newPlayer = {
        id: k,
        name,
        points
      };

      newPlayers.push(newPlayer);
    }

    newPlayers.sort((a, b) => b.points - a.points);
    newPlayers = newPlayers.splice(0, 10);

    this.setState({
      players: newPlayers
    });
  };

  gotError = err => console.log('Error ', err);

  componentDidMount() {
    const { firebase } = this.state;
    firebase.getDataFromFireBase(this.gotData, this.gotError);
  }

  render() {
    const { firebase } = this.state;
    let { players, addScore, numberOfClickToAdd } = this.state;
    return (
      <Container fluid={true} className="text-center">
        <Col md="12" sm="12" xs="12">
          <MainPage>
            <Header addScore={addScore} />
            <Home
              addScore={addScore}
              players={players}
              addNewScore={this.addNewScore}
              changeSite={this.changeSite}
              numberOfClickToAdd={numberOfClickToAdd}
              addScoreToFireBase={firebase.addScoreToFireBase}
              changeAddScore={this.changeAddScore}
            />
          </MainPage>
          <Footer players={players} />
        </Col>
      </Container>
    );
  }
}

export default App;

const MainPage = styled.div`
  background-color: #eee;
  padding-top: 48px;
  padding-bottom: 48px;
  text-align: center;
`;

export const Title = styled.h1`
  font-size: 4.5rem;
  font-weight: 300;
`;

export const Paragraf = styled.p`
  font-size: 1.4rem;
  font-weight: 200;
  margin-bottom: 15px;
`;
