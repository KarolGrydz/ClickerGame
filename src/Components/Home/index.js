import React from 'react';
import Main from './Main';
import AddScore from './AddScore';

const Home = ({
  players,
  addNewScore,
  numberOfClickToAdd,
  changeSite,
  addScoreToFireBase,
  changeAddScore,
  addScore
}) => {
  return (
    <div>
      {addScore ? (
        <AddScore
          numberOfClickToAdd={numberOfClickToAdd}
          addScoreToFireBase={addScoreToFireBase}
          changeAddScore={changeAddScore}
          changeSite={changeSite}
        />
      ) : (
        <Main
          players={players}
          addNewScore={addNewScore}
          changeSite={changeSite}
        />
      )}
    </div>
  );
};

export default Home;
