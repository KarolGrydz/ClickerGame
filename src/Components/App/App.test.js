import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import App from './index';
import Header from '../Header/index';
import Home from '../Home/index';
import Footer from '../Footer/index';

describe('<App />', () => {
  it('should be defined', () => {
    expect(App).toBeDefined();
  });
  it('should render without crashing', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.containsMatchingElement(<Header />)).toBeTruthy();
    expect(wrapper.containsMatchingElement(<Home />)).toBeTruthy();
    expect(wrapper.containsMatchingElement(<Footer />)).toBeTruthy();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
