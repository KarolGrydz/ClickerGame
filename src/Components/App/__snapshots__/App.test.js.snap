// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render without crashing 1`] = `
<Container
  className="text-center"
  fluid={true}
  tag="div"
>
  <Col
    md="12"
    sm="12"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="12"
  >
    <styled.div>
      <AppHeader
        addScore={false}
      />
      <Home
        addNewScore={[Function]}
        addScore={false}
        addScoreToFireBase={[Function]}
        changeSite={[Function]}
        numberOfClickToAdd={0}
        players={Array []}
      />
    </styled.div>
    <Footer
      players={Array []}
    />
  </Col>
</Container>
`;
